lldb/include/lldb/Core/ValueObject.h

```diff
@@ -441,6 +441,19 @@ class ValueObject {
 
   virtual int64_t GetValueAsSigned(int64_t fail_value, bool *success = nullptr);
 
+  llvm::APSInt GetValueAsAPSInt();
+
+  llvm::APFloat GetValueAsFloat();
+
+  bool GetValueAsBool();
```

Comment:
I would prefer to have GetValueAsBool().  Otherwise I will need to do integer-to-bool conversions in the places where I call this function; also it makes the code more clear, readable and maintainable.  But if you feel very strongly about this, I can fall back on GetValueAsAPSInt.